src/main/java/mysutong/Parser.java

```diff
@@ -0,0 +1,92 @@
+                    ui.showMessage(tasks.getTask(markIndex).toString());
+                    ui.showLine();
+                    storage.save(tasks);
+                    break;
+
+                case "unmark":
+                    int unmarkIndex = Integer.parseInt(inputs[1]) - 1;
+                    tasks.getTask(unmarkIndex).unmark();
+                    ui.showLine();
+                    ui.showMessage("OK, I've marked this task as not done yet:");
+                    ui.showMessage(tasks.getTask(unmarkIndex).toString());
+                    ui.showLine();
+                    storage.save(tasks);
+                    break;
+
+                case "todo":
+                    tasks.addTask(new Todo(inputs[1]));
+                    ui.showLine();
+                    ui.showMessage("Got it. I've added this task:");
+                    ui.showMessage(tasks.getTask(tasks.getTasks().size() - 1).toString());
+                    ui.showLine();
+                    storage.save(tasks);
+                    break;
+
+                case "deadline":
+                    String[] deadlineDetails = inputs[1].split("/by");
+                    tasks.addTask(new Deadline(deadlineDetails[0].trim(), LocalDateTime.parse(deadlineDetails[1].trim(), DateTimeFormatter.ofPattern("d/M/yyyy HHmm"))));
+                    ui.showLine();
+                    ui.showMessage("Got it. I've added this task:");
+                    ui.showMessage(tasks.getTask(tasks.getTasks().size() - 1).toString());
+                    ui.showLine();
+                    storage.save(tasks);
+                    break;
+
+                case "event":
+                    String[] eventDetails = inputs[1].split("/from|/to");
+                    tasks.addTask(new Event(eventDetails[0].trim(), LocalDateTime.parse(eventDetails[1].trim(), DateTimeFormatter.ofPattern("d/M/yyyy HHmm")), LocalDateTime.parse(eventDetails[2].trim(), DateTimeFormatter.ofPattern("d/M/yyyy HHmm"))));
+                    ui.showLine();
+                    ui.showMessage("Got it. I've added this task:");
+                    ui.showMessage(tasks.getTask(tasks.getTasks().size() - 1).toString());
+                    ui.showLine();
+                    storage.save(tasks);
+                    break;
+
+                case "delete":
+                    int deleteIndex = Integer.parseInt(inputs[1]) - 1;
+                    Task removedTask = tasks.getTask(deleteIndex);
+                    tasks.removeTask(deleteIndex);
+                    ui.showLine();
+                    ui.showMessage("Noted. I've removed this task:");
+                    ui.showMessage(removedTask.toString());
+                    ui.showLine();
+                    storage.save(tasks);
+                    break;
+
+                default:
+                    throw new UnknownCommandException("I'm sorry, but I don't know what that means.");
+            }
```

Comment:
```suggestion
            switch (command) {
            case "bye":
                ui.showGoodbye();
                System.exit(0);
                break;
            case "list":
                ui.showTaskList(tasks);
                break;
            case "mark":
                int markIndex = Integer.parseInt(inputs[1]) - 1;
                tasks.getTask(markIndex).markAsDone();
                ui.showLine();
                ui.showMessage("Nice! I've marked this task as done:");
                ui.showMessage(tasks.getTask(markIndex).toString());
                ui.showLine();
                storage.save(tasks);
                break;
            case "unmark":
                int unmarkIndex = Integer.parseInt(inputs[1]) - 1;
                tasks.getTask(unmarkIndex).unmark();
                ui.showLine();
                ui.showMessage("OK, I've marked this task as not done yet:");
                ui.showMessage(tasks.getTask(unmarkIndex).toString());
                ui.showLine();
                storage.save(tasks);
                break;
            case "todo":
                tasks.addTask(new Todo(inputs[1]));
                ui.showLine();
                ui.showMessage("Got it. I've added this task:");
                ui.showMessage(tasks.getTask(tasks.getTasks().size() - 1).toString());
                ui.showLine();
                storage.save(tasks);
                break;
            case "deadline":
                String[] deadlineDetails = inputs[1].split("/by");
                tasks.addTask(new Deadline(deadlineDetails[0].trim(), LocalDateTime.parse(deadlineDetails[1].trim(), DateTimeFormatter.ofPattern("d/M/yyyy HHmm"))));
                ui.showLine();
                ui.showMessage("Got it. I've added this task:");
                ui.showMessage(tasks.getTask(tasks.getTasks().size() - 1).toString());
                ui.showLine();
                storage.save(tasks);
                break;
            case "event":
                String[] eventDetails = inputs[1].split("/from|/to");
                tasks.addTask(new Event(eventDetails[0].trim(), LocalDateTime.parse(eventDetails[1].trim(), DateTimeFormatter.ofPattern("d/M/yyyy HHmm")), LocalDateTime.parse(eventDetails[2].trim(), DateTimeFormatter.ofPattern("d/M/yyyy HHmm"))));
                ui.showLine();
                ui.showMessage("Got it. I've added this task:");
                ui.showMessage(tasks.getTask(tasks.getTasks().size() - 1).toString());
                ui.showLine();
                storage.save(tasks);
                break;
            case "delete":
                int deleteIndex = Integer.parseInt(inputs[1]) - 1;
                Task removedTask = tasks.getTask(deleteIndex);
                tasks.removeTask(deleteIndex);
                ui.showLine();
                ui.showMessage("Noted. I've removed this task:");
                ui.showMessage(removedTask.toString());
                ui.showLine();
                storage.save(tasks);
                break;
            default:
                throw new UnknownCommandException("I'm sorry, but I don't know what that means.");
          }
```

The coding standard mentions that `switch` statements should not have indentation for `case` clauses. Default formatting for your IDE may have conflicted with this.